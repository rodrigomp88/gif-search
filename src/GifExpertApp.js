import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>Gif Search</h2>
      <p>Escriba para visualizar lo que desee ver</p>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid category={category} key={category} />
        ))}
      </ol>
      <div class="footer">
        Made with ‚ù§, by <a href="https://github.com/rodrigomp88">Rodrigo</a>
      </div>
    </>
  );
};
