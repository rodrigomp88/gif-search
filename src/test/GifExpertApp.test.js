import React from "react";
import { shallow } from "enzyme/build";
import { GifExpertApp } from "../GifExpertApp";

describe("Pruebas en GifExpertApp", () => {
  test("debe msotrar correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lisra de categorias', () => {
      const categories = ['Dragon Ball', 'One Punch'];
      const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

      expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
  
});
