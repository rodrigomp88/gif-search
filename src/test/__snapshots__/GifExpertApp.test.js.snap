// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe msotrar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
